hometask7.py

```diff
@@ -0,0 +1,105 @@
+# Task 1
+import json
+
+
+try:
+    with open('json_phonebook.json', 'r') as file:
+        data = file.read()
+        phonebook = json.loads(data)
+except FileNotFoundError:
+    # if file does not exist
+    phonebook = {}
+
+while True:
+    user_input = input('Enter a command:')
+    split_input = user_input.split()
+    command = split_input[0]
+
+
+    if command == 'stats':
+        print(len(phonebook))
+
+
+    elif command == 'add':
+        name = input('Enter a name:')
+        if not name in phonebook.keys():
+            number = input('Enter a number:')
+            phonebook[name] = number
+            with open('json_phonebook.json', 'w+') as file:
+                content = json.dumps(phonebook)
+                file.write(content)
+            print('New contact was added')
+    
+
+    elif command == 'del':
+        key = input('Enter some name: ')
+        if phonebook.get(key):
+            del phonebook[key]
+            with open('json_phonebook.json', 'w') as file:
+                data = json.dumps(phonebook)
+                file.write(data)
+            print("Record was deleted")
+        else:
+            print('Such record never existed')
+
+
+
+    elif command == 'list':
+        list = (phonebook.items())
+        print(list)
+
+
+    elif command == 'show':
+        print(phonebook)
+        name = input('Enter a name:')
+        if name not in phonebook.keys():
+            print("Fault. There is no such name in a phonebook.")
+        else:
+            print(phonebook[name])
+
+
+    elif command == 'end':
+        break
+
+
+
+# Task 2
+
+import time
+import os
+
+f = open('log1.txt', "w+")
```

Comment:
You should open this file in the place where you use it. Move this line between 78 and 79 lines. And also close the file there, between 80 and 81 lines

Reply:
Done

Reply:
Done